src/elfpy/markets.py

```diff
@@ -216,50 +294,71 @@ def update_spot_price(self):
             share_price=self.share_price,
             time_remaining=time_utils.stretch_time(self.token_duration, self.time_stretch_constant),
         )
+        self.rate = price_utils.calc_apr_from_spot_price(self.spot_price, self.token_duration)
+
+    def swap(self, trade_details):
```

Comment:
Why does this function exist? It is only called in `_open_short` and the conditional is already fixed in `trade_and_update`, so it's unnecessary.